### ISSUE TITLE
[VersionControl] Crash getting file history (Git)

### ISSUE THREAD

**Issue by jsuarezruiz** [mono/monodevelop#7105]
IDE crash getting the history of a file.

```
ERROR [2019-02-08 10:33:05Z]: Error while getting revision text
System.NullReferenceException: Object reference not set to an instance of an object
  at (wrapper managed-to-native) LibGit2Sharp.Core.NativeMethods.git_object_lookup(LibGit2Sharp.Core.git_object*&,LibGit2Sharp.Core.git_repository*,LibGit2Sharp.Core.GitOid&,LibGit2Sharp.Core.GitObjectType)
  at LibGit2Sharp.Core.Proxy.git_object_lookup (LibGit2Sharp.Core.Handles.RepositoryHandle repo, LibGit2Sharp.ObjectId id, LibGit2Sharp.Core.GitObjectType type) [0x00008] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/external/libgit2sharp/LibGit2Sharp/Core/Proxy.cs:1465
  at LibGit2Sharp.Core.ObjectSafeWrapper..ctor (LibGit2Sharp.ObjectId id, LibGit2Sharp.Core.Handles.RepositoryHandle handle, System.Boolean allowNullObjectId) [0x00044] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/external/libgit2sharp/LibGit2Sharp/Core/ObjectSafeWrapper.cs:21
  at LibGit2Sharp.Core.Proxy.git_tree_entry_bypath (LibGit2Sharp.Core.Handles.RepositoryHandle repo, LibGit2Sharp.ObjectId id, System.String treeentry_path) [0x00001] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/external/libgit2sharp/LibGit2Sharp/Core/Proxy.cs:3263
  at LibGit2Sharp.Tree.RetrieveFromPath (System.String relativePath) [0x00013] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/external/libgit2sharp/LibGit2Sharp/Tree.cs:59
  at LibGit2Sharp.Tree.get_Item (System.String relativePath) [0x00001] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/external/libgit2sharp/LibGit2Sharp/Tree.cs:49
  at LibGit2Sharp.Commit.get_Item (System.String relativePath) [0x00001] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/external/libgit2sharp/LibGit2Sharp/Commit.cs:61
  at MonoDevelop.VersionControl.Git.GitRepository.GetBlob (LibGit2Sharp.Commit c, MonoDevelop.Core.FilePath file) [0x00001] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/addins/VersionControl/MonoDevelop.VersionControl.Git/MonoDevelop.VersionControl.Git/GitRepository.cs:1179
  at MonoDevelop.VersionControl.Git.GitRepository.GetCommitTextContent (LibGit2Sharp.Commit c, MonoDevelop.Core.FilePath file) [0x00001] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/addins/VersionControl/MonoDevelop.VersionControl.Git/MonoDevelop.VersionControl.Git/GitRepository.cs:1185
  at MonoDevelop.VersionControl.Git.GitRepository.OnGetTextAtRevision (MonoDevelop.Core.FilePath repositoryPath, MonoDevelop.VersionControl.Revision revision) [0x00008] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/addins/VersionControl/MonoDevelop.VersionControl.Git/MonoDevelop.VersionControl.Git/GitRepository.cs:1148
  at MonoDevelop.VersionControl.Repository.GetTextAtRevision (MonoDevelop.Core.FilePath repositoryPath, MonoDevelop.VersionControl.Revision revision) [0x00001] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl/Repository.cs:908
  at MonoDevelop.VersionControl.Views.LogWidget+<>c__DisplayClass36_1.<HandleTreeviewFilesTestExpandRow>b__0 (System.Object <p0>) [0x00015] in /Users/jsuarezruiz/Projects/MonoDevelop/monodevelop/main/src/addins/VersionControl/MonoDevelop.VersionControl/MonoDevelop.VersionControl.Views/LogWidget.cs:405
ERROR [2019-02-08 10:33:05Z]: Error while getting revision text.. The file may not be part of the working copy.
```

Regression #7078.

> VS bug [#789302](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/789302)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.0.0.1594 (master)

Pull Request #7078 merged by: Vsevolod Kukol
Author: mono
Commit: e9006c97e60bc51b1fe82dec9f80c9eb4f57b7cc (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.0.0.2351 (release-8.0)

Pull Request #7078 merged by: Vsevolod Kukol
Author: mono
Commit: e9006c97e60bc51b1fe82dec9f80c9eb4f57b7cc (xamarin/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.0.0.2351 (release-8.0)

Pull Request #7078 merged by: Vsevolod Kukol
Author: mono
Commit: e9006c97e60bc51b1fe82dec9f80c9eb4f57b7cc (xamarin/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.6.0.802 (master)

Pull Request #7078 merged by: Vsevolod Kukol
Author: mono
Commit: e9006c97e60bc51b1fe82dec9f80c9eb4f57b7cc (xamarin/monodevelop)
